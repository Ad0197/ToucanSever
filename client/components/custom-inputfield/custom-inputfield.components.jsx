import React from "react";
import { CustomInputFieldContainer, CustomInputLabel, CustomInputText } from "./custom-inputfield.styles";

const CustomInputField = ({ ref, onChange, value, placeholder = "Write here...", type = "text", label, name }) => (
  <CustomInputFieldContainer>
    <CustomInputText {...{ onChange, ref, value, placeholder, type, name }} />
  </CustomInputFieldContainer>
);

export default CustomInputField;
