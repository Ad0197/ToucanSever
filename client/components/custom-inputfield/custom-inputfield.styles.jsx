import styled from "styled-components";

export const CustomInputFieldContainer = styled.div`
  width: 100%;
  background-color: ${({ theme }) => theme.light};
  border-radius: 50px;
  padding: 10px 20px;
  margin: 10px;
  &:focus-within {
    -webkit-box-shadow: 0px 0px 15px 5px rgba(28, 10, 160, 0.3);
    box-shadow: 0px 0px 15px 5px rgba(28, 10, 160, 0.3);
  }
`;

export const CustomInputText = styled.input`
  width: 100%;
  background-color: transparent;
  border: none;
  outline: none;
`;

export const CustomInputLabel = styled.label``;

//   -webkit-box-shadow: 0px 0px 32px -1px rgba(0, 0, 0, 0.42);
//   box-shadow: 0px 0px 32px -1px rgba(0, 0, 0, 0.42);
