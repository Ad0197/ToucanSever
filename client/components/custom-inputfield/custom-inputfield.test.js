import CustomInputField from "./custom-inputfield.components";
import { CustomInputText } from "./custom-inputfield.styles";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("CustomInputField component", () => {
  it("should has a onChange props", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<CustomInputField onChange={onChange} />);
    wrapper
      .find(CustomInputText)
      .at(0)
      .simulate("change", { target: { name: "", value: "TEST" } });
    expect(onChange).toHaveBeenCalled();
  });

  it("should has a value props, placeholder", () => {
    const value = "TEST VALUE",
      placehoder = "PLACEHODER VALUE",
      name = "name";
    const wrapper = shallow(<CustomInputField value={value} placeholder={placehoder} name={name} />);
    expect(wrapper.find(CustomInputText).at(0).prop("value")).toBe(value);
    expect(wrapper.find(CustomInputText).at(0).prop("placeholder")).toBe(placehoder);
    expect(wrapper.find(CustomInputText).at(0).prop("name")).toBe(name);
  });

  it("should match with snapshop", () => {
    const wrapper = shallow(<CustomInputField {...{ placeholder: "", onChange: () => {}, name: "name" }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
