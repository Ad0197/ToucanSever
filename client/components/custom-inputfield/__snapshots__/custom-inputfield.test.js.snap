// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInputField component should match with snapshop 1`] = `
<custom-inputfieldstyles__CustomInputFieldContainer>
  <custom-inputfieldstyles__CustomInputText
    name="name"
    onChange={[Function]}
    placeholder=""
    type="text"
  />
</custom-inputfieldstyles__CustomInputFieldContainer>
`;
